site-src/guides/implementers-guide.md

```diff
@@ -0,0 +1,367 @@
+time, things will probably work, but if it doesn't work, it will most likely 
+break in weird ways.
+
+Try your best to ensure that the bundle version doesn't roll backwards. It's safer.
+
+Implementations SHOULD also handle the Gateway API CRDs _not_ being present in
+the cluster without crashing or panicing. Exiting with a clear fatal error is
+acceptable in this case, as is disabling Gateway API support even if enabled in
+configuration.
+
+Practically, for implementations using tools like `controller-runtime` or
+similar tooling, they may need to check for the _presence_ of the CRDs by 
+getting the list of installed CRDs before attempting to watch those resources.
+(Note that this will require the implementation to have `read` access to those
+resources though.)
+
+#### Automatic CRD installation
+
+Automatic CRD installation also includes automatic installation mechanisms such
+as Helm, if the CRDs are included in a Helm chart with the implementation's
+installation.
+
+CRD definitions MAY be installed automatically by implementations, and if they do,
+they MUST have a way to ensure:
+
+- there are no other Gateway API CRDs installed in the cluster before starting, or
+- that the CRD definitions are only installed if they are a higher bundle version
+  than any existing Gateway API CRDs
+
+This avoids problems if another implementation is also installed in the cluster
+and expects a higher version of the CRDs to be installed.
+
+If the implementation can guarantee that no other implementation will interact
+with the cluster, then it MAY automatically install a relevant version of the CRDs.
+
+The ideal method for an automatic installation would require the implementation
+to:
+
+- Check if there are any Gateway API CRDs installed in the cluster.
+- If not, install its most compatible version of the CRDs.
+- If so, only install its version of the CRDs if the bundle version is higher
+  than the existing one.
+
+
+Because of our backwards compatibility guarantees, it's also safe for a controller
+to flip the install channel between "standard" and "experimental", although
+implementations MUST NOT do this without consulting the implementation owner.
+
+Automatic CRD installation has the advantage that there is less for the
+implementation user to do; any required version checking can be performed by
+code instead of by the cluster admin.
+
+#### Manual CRD installation
+
+Manual CRD installation has the advantage that the implementer needs to maintain
+less code; however, it pushes the responsibility for correctly managing the
+Gateway API CRDs back to the cluster admin, who may not have as much context
+as is provided here.
+
+Implementations MAY require the installation to be done manually; if so, the
+installation instructions SHOULD include commands to check if there are any other
+CRDs installed already and verify that the installation will not be a downgrade.
```

Comment:
As a followup maybe we should provide a common command to do this.

Istio, for instance, has an example of a suboptimal approach: https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#setup

Reply:
Agreed, this is a great followup. I'll create an issue a bit later for it.